internal/zinc-persist/src/main/scala/sbt/internal/inc/FileBasedStore.scala

```diff
@@ -6,48 +6,67 @@ package internal
 package inc
 
 import java.io._
-import java.util.zip.ZipEntry
+import java.util.zip.{ ZipInputStream, ZipEntry }
 import sbt.io.{ IO, Using }
+import xsbti.SafeLazy
 import xsbti.compile.{ CompileAnalysis, MiniSetup }
+import xsbti.api.Companions
+import scala.util.control.Exception.allCatch
 
 object FileBasedStore {
-  def apply(file: File): AnalysisStore = new AnalysisStore {
+  private val analysisFileName = "inc_compile.txt"
+  private val companionsFileName = "api_companions.txt"
 
-    private val entryFileName = "cache.txt"
+  def apply(file: File): AnalysisStore = new FileBasedStore0(file)
+
+  private final class FileBasedStore0(file: File) extends AnalysisStore {
```

Comment:
The whole point is that I'm mimicking the `object FileBaseStore` so I think that's fine. It's only so the stack trace looks nicer.